src/credentials/DeviceAttestationVerifier.h

```diff
@@ -0,0 +1,126 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#pragma once
+
+#include <crypto/CHIPCryptoPAL.h>
+#include <lib/core/CHIPError.h>
+#include <lib/support/Span.h>
+
+namespace chip {
+namespace Credentials {
+
+enum class AttestationVerificationResult : uint16_t
+{
+    kSuccess = 0,
+
+    kPaaUntrusted        = 100,
+    kPaaNotFound         = 101,
+    kPaaExpired          = 102,
+    kPaaSignatureInvalid = 103,
+    kPaaRevoked          = 104,
+    kPaaFormatInvalid    = 105,
+
+    kPaiExpired           = 200,
+    kPaiSignatureInvalid  = 201,
+    kPaiRevoked           = 202,
+    kPaiFormatInvalid     = 203,
+    kPaiVendorIdMismatch  = 204,
+    kPaiAuthorityNotFound = 205,
+
+    kDacExpired           = 300,
+    kDacSignatureInvalid  = 301,
+    kDacRevoked           = 302,
+    kDacFormatInvalid     = 303,
+    kDacVendorIdMismatch  = 304,
+    kDacProductIdMismatch = 305,
+    kDacAuthorityNotFound = 306,
+
+    kFirmwareInformationMismatch = 400,
+    kFirmwareInformationMissing  = 401,
+
+    kCertificationDeclarationMissing = 500,
+
+    kNonceMismatch,
+
+    kInvalidSignature,
+
+    kNoMemory,
```

Comment:
Suggest putting those in a range and allocating numbers, so it's easy to match during debug.

For kInvalidSignature, please rename to `kInvalidSignatureFormat`.

Please also add `kAttestationSignatureInvalid` (see my other comment referencing it)

Reply:
done